x-pack/plugins/apm/server/routes/data_view/create_static_data_view.ts

```diff
@@ -74,27 +103,21 @@ export async function createStaticDataView({
   });
 }
 
-// force an overwrite of the data view if the data view has been changed
-async function getForceOverwrite({
-  savedObjectsClient,
+// only create data view if it doesn't exist or was changed
+async function getShouldCreateOrUpdate({
+  dataViewService,
   apmDataViewTitle,
 }: {
-  savedObjectsClient: InternalSavedObjectsClient;
+  dataViewService: DataViewsService;
   apmDataViewTitle: string;
 }) {
   try {
-    const existingDataView =
-      await savedObjectsClient.get<ApmDataViewAttributes>(
-        'index-pattern',
-        APM_STATIC_DATA_VIEW_ID
-      );
-
-    // if the existing data view does not matches the new one, force an update
-    return existingDataView.attributes.title !== apmDataViewTitle;
+    const existingDataView = await dataViewService.get(APM_STATIC_DATA_VIEW_ID);
+    return existingDataView.title !== apmDataViewTitle;
```

Comment:
is this logic still good enough? if it's always going to be `'APM UI'`? Previously it was the index target, so if that changed the data view would be updated as well. If we now change the index target, the data view won't get updated IIUC?

Reply:
So to answer your question: yes, this logic still holds (I tested it :D )

Reply:
ah, woops! definitely confusing 😄